modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -3028,26 +3029,31 @@ else if (sqlType == Types.VARCHAR) {
 	}
 
 	private void _setExternalReferenceCode(
-			ObjectEntry objectEntry, Map<String, Serializable> values)
+			long objectDefinitionId, ObjectEntry objectEntry,
+			Map<String, Serializable> values)
 		throws PortalException {
 
 		for (Map.Entry<String, Serializable> entry : values.entrySet()) {
 			if (StringUtil.equals(entry.getKey(), "externalReferenceCode")) {
 				String externalReferenceCode = String.valueOf(entry.getValue());
 
 				if (Validator.isNull(externalReferenceCode)) {
-					externalReferenceCode = String.valueOf(
-						objectEntry.getObjectEntryId());
+					break;
```

Comment:
@p-albuquerque aren't we changing the expected behavior by removing this code?

Reply:
The behavior that it code represent, I believe isn't right, there is an ambiguous behavior around that:
The **ERC** is assigned with **Uuid** value, when we add a _ObjectEntry_ via UI or API(in that case, if we don't explicit a value for ERC).
But, if we are receiving something like this following **valuesMap** of an _ObjectEntry_, the default value assigned in ERC is **objectEntryId** instead **Uuid** :
![ERCIssue](https://user-images.githubusercontent.com/79872134/227553040-2c95b742-eaa3-454e-b627-5573ad2964bc.png)

Reply:
besides that, if this scenarios occour while an update, the current ERC will be replaced to **objectEntryId**. With that changes, while update, if some map like this one above will received. The current ERC will be untouched.